AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,192 @@
+            case HubPayload.EventName.DataStore.modelSynced:
+                guard let modelSyncedEventPayload = payload.data as? ModelSyncedEvent else {
+                    XCTFail("Couldn't cast payload data as ModelSyncedEvent")
+                    return
+                }
+                modelSyncedEventPayloads.append(modelSyncedEventPayload)
+
+                if modelSyncedEventPayloads.count == 2 {
+                    XCTAssertTrue(modelSyncedEventPayloads[0] == self.expectedModelSyncedEventPayloads[0])
+                    XCTAssertTrue(modelSyncedEventPayloads[1] == self.expectedModelSyncedEventPayloads[1])
+                    modelSyncedReceived.fulfill()
+                }
+            case HubPayload.EventName.DataStore.syncQueriesReady:
+                syncQueriesReadyReceived.fulfill()
+            default:
+                break
+            }
+        }
+
+        let syncableModelTypes = ModelRegistry.models.filter { $0.schema.isSyncable }
+
+        reconciliationQueue = MockAWSIncomingEventReconciliationQueue(modelTypes: syncableModelTypes,
+                                                                          api: nil,
+                                                                          storageAdapter: nil,
+                                                                          auth: nil)
+
+        initialSyncOrchestrator = MockAWSInitialSyncOrchestrator(dataStoreConfiguration: .default,
+                                                                 api: nil,
+                                                                 reconciliationQueue: nil,
+                                                                 storageAdapter: nil)
+
+        syncEventEmitter = SyncEventEmitter(initialSyncOrchestrator: initialSyncOrchestrator,
+                                                reconciliationQueue: reconciliationQueue)
+
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.started(modelType: Post.self, syncType: .fullSync))
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.mutationSync(anyPostMutationSync))
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.finished(modelType: Post.self))
+
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.started(modelType: Comment.self, syncType: .fullSync))
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.mutationSync(anyCommentMutationSync))
+        initialSyncOrchestrator?.initialSyncOrchestratorTopic.send(.finished(modelType: Comment.self))
+
+        reconciliationQueue?.incomingEventSubject.send(.mutationEventApplied(postMutationEvent))
+        reconciliationQueue?.incomingEventSubject.send(.mutationEventApplied(commentMutationEvent))
+
+        waitForExpectations(timeout: 1)
+        syncEventEmitter = nil
+        listener.cancel()
+    }
+}
+
+extension ModelSyncedEvent {
+    static func == (lhs: ModelSyncedEvent, rhs: ModelSyncedEvent) -> Bool {
```

Comment:
better place can be in `EquatableExtensions.swift`

Reply:
Moved. I didn't know there is `EquatableExtensions.swift`